File: src/main/java/gr/aueb/cf/inspirecraft/model/Creator.java
package gr.aueb.cf.inspirecraft.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.List;
import java.util.UUID;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "creators")
public class Creator extends AbstractEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String uuid;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name ="user_id")
    private User user;

    @OneToMany(mappedBy = "creator", cascade = CascadeType.ALL)
    private List<Product> products;

    @PrePersist
    public void initializeUUID(){
        if (uuid == null) uuid = UUID.randomUUID().toString();
    }
}

File: inspirecraft-frontend/src/app/interfaces/creator.ts
import { User } from './user';

export interface Creator {
  id: number;
  uuid: string;
  products?: Product[]; // Προϊόντα που έχει ο Creator
  user: User;
}

export interface Product {
  id: number;
  name: string;
  description: string;
  imageUrl: string;
  creatorId: number; // Σύνδεση με τον Creator
}
